### ISSUE TITLE
Storing and using the refresh token

### ISSUE THREAD

**Issue by MarkZamoyta** [aspnet/Docs#9314]
This works great with the access token, but it has a short lifespan so I need to put the refresh token to use.  I see some code at the Google .Net API site that uses a class called GoogleAuthorizationCodeFlow, which looks like it manages the refresh token when needed to get a new access token.  The problem is it doesn't work with Identity / EF and the whole .AddGoogle() extension method.  
Is there any sample code that checks for an expired access token and uses the refresh token to acquire a new one?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 64faf33b-2511-5e8f-c1b0-94f32dc07fa4
* Version Independent ID: 2697abc8-7727-2eba-74a5-fbbc8a7c80ff
* Content: [Persist additional claims and tokens from external providers in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/additional-claims?view=aspnetcore-2.1)
* Content Source: [aspnetcore/security/authentication/social/additional-claims.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/security/authentication/social/additional-claims.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
TL;DR @HaoK We don't address the refresh tokens scenario. It was very briefly mentioned in the [issue for the topic (aspnet/Docs \#3744)](https://github.com/aspnet/Docs/issues/3744#issue-243489148), but it didn't come up thereafter and wasn't part of the AuthSamples.ExternalClaims app upon which this sample was based.

@MarkZamoyta ...

Good point ... We don't address the subject outside of `SaveTokens` and `StoreTokens`. We **_don't_** describe setting `AccessType` to `offline`, for example. The sample here was based on the AuthSamples.ExternalClaims sample, which also doesn't address it.

However, I did find an issue that might help, see :point_right: https://github.com/aspnet/Security/issues/1668

I only looked at one issue on a quick search. I recommend searching that repo's open **and closed** issues to see what else you can turn up there.

If you have questions about anything there on a closed issue, open a new issue at https://github.com/aspnet/Security/issues. They don't respond on closed issues other than to tell you to please open a new issue. 😉

I'll discuss this further with the engineers. Thanks for opening this issue.

**Comment by Rick-Anderson** [comment]
Thanks for contacting us.
We don’t have the resources to invest in this area, so we are closing the issue. Should your request generate enough  👍 responses, we’ll reconsider.
